order/bounded_lattice.lean

```diff
@@ -282,6 +289,10 @@ lemma some_eq_coe (a : α) : (some a : with_bot α) = (↑a : with_bot α) := rf
 theorem coe_eq_coe {a b : α} : (a : with_bot α) = b ↔ a = b :=
 by rw [← option.some.inj_eq a b]; refl
 
+@[priority 0]
+instance has_lt [has_lt α] : has_lt (with_bot α) :=
+{ lt := λ o₁ o₂ : option α, o₂.is_some ∧ ∀ a ∈ o₁, ∃ b ∈ o₂, a < b }
```

Comment:
I'm not a big fan of this definition, moreso because it conflicts with the partial order definition. You don't need the `is_some` part in there either, just write the quantifiers the other way around - `∃ b ∈ o₂, ∀ a ∈ o₁, a < b`

Reply:
Yes, I don't love that either. Would you prefer it if I made it a definition and only locally make it an instance I when I need it?

I'll make that change.

Reply:
I think it should be also used as the `lt` field of `partial_order`, so there is no conflict

Reply:
Ooooh! Somehow I overlooked that. I thought `has_lt` was derived. That's much better.